### PR TITLE
etcd: Provide endpoints and config when connecting

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6071)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -223,7 +223,10 @@ func newEtcdClient(config *client.Config, cfgPath string) (BackendOperations, er
 	if err != nil {
 		return nil, err
 	}
-	log.Info("Waiting for etcd client to be ready")
+	log.WithFields(logrus.Fields{
+		"endpoints": config.Endpoints,
```
Comment (ianvernon) [inline pkg/kvstore/etcd.go:222]: config can be nil - this would panic if it is nil.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: pkg/kvstore/etcd.go (hunk 1 above)
Comment (tgraf) [reply pkg/kvstore/etcd.go:222]: The branch in that function is actually dead code. The only caller of `newEtcdClient()` does:
```
          if e.config == nil {
                  if !endpointsSet && !configSet {
                          return nil, fmt.Errorf("invalid etcd configuration, %s or %s must be specified",
                                  EtcdOptionConfig, addrOption)
                  }

                  e.config = &client.Config{}
```

Comment (tgraf) [reply pkg/kvstore/etcd.go:222]: I will remove the dead code

**Comment by tgraf** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**